src/models/message.ts

```diff
@@ -257,23 +229,60 @@ export class MessageModel extends DocumentDbModel<
    */
   public findMessages(
     fiscalCode: FiscalCode
-  ): DocumentDbUtils.IResultIterator<RetrievedMessageWithContent> {
-    return DocumentDbUtils.queryDocuments(
-      this.dbClient,
-      this.collectionUri,
-      {
-        parameters: [
-          {
-            name: "@fiscalCode",
-            value: fiscalCode
-          }
-        ],
-        query: `SELECT * FROM m WHERE m.${MESSAGE_MODEL_PK_FIELD} = @fiscalCode`
-      },
-      fiscalCode
+  ): TaskEither<
+    CosmosErrors,
+    AsyncIterator<ReadonlyArray<t.Validation<RetrievedMessage>>>
+  > {
+    const iterator = this.getQueryIterator({
+      parameters: [
+        {
+          name: "@fiscalCode",
+          value: fiscalCode
+        }
+      ],
+      query: `SELECT * FROM m WHERE m.${MESSAGE_MODEL_PK_FIELD} = @fiscalCode`
+    })[Symbol.asyncIterator]();
+
+    return fromEitherT(
+      tryCatch2v(
+        () => iterator,
+        _ => CosmosErrorResponse(_ as ErrorResponse)
+      )
     );
   }
 
+  /**
+   * @deprecated use getQueryIterator + asyncIterableToArray
+   */
+  public findAllByQuery(
+    query: string | SqlQuerySpec,
+    options?: FeedOptions
+  ): TaskEither<
+    CosmosErrors,
+    Option<ReadonlyArray<RetrievedMessageWithoutContent>>
+  > {
+    const fetchAll = this.container.items.query<RetrievedMessageWithoutContent>(
+      query,
+      options
+    ).fetchAll;
+    return tryCatchT<CosmosErrors, PromiseType<ReturnType<typeof fetchAll>>>(
+      () => fetchAll(),
+      _ => CosmosErrorResponse(_ as ErrorResponse)
+    )
+      .map(_ => fromNullable(_.resources))
+      .chain(_ =>
+        _.isSome()
+          ? fromEitherT(
+              array.sequence(either)(
+                _.value.map(RetrievedMessageWithoutContent.decode)
+              )
+            )
+              .map(some)
+              .mapLeft(CosmosDecodingError)
+          : fromEitherT(right(none))
+      );
```

Comment:
this should go into the parent model since it's a generic method. can we rewrite findMessages using this one?

Reply:
It' s deprecated, cause we have to use `findMessages` using `AsyncIterator`. Maybe It can be an other option to implement this method in a generic way for other purposes @cloudify